Empty automation sentence trigger crashes entire recognition engine

### The problem

Having a sentence trigger with no sentence in an automation makes the entire default conversation engine fail.

### What version of Home Assistant Core has the issue?

core-2023.11.1

### What was the last working version of Home Assistant Core?

_No response_

### What type of installation are you running?

Home Assistant OS

### Integration causing the issue

conversation

### Link to integration documentation on our website

https://www.home-assistant.io/integrations/conversation/

### Diagnostics information

_No response_

### Example YAML snippet

```yaml
alias: New automation
description: ""
trigger:
  - platform: conversation
    command: ""
condition: []
action: []
mode: single
```


### Anything in the logs that might be useful for us?

```txt
Traceback (most recent call last):
  File "/workspaces/ha-core/homeassistant/components/assist_pipeline/pipeline.py", line 938, in recognize_intent
    conversation_result = await conversation.async_converse(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/ha-core/homeassistant/components/conversation/__init__.py", line 467, in async_converse
    result = await agent.async_process(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/ha-core/homeassistant/components/conversation/default_agent.py", line 202, in async_process
    if trigger_result := await self._match_triggers(user_input.text):
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/ha-core/homeassistant/components/conversation/default_agent.py", line 697, in _match_triggers
    self._rebuild_trigger_intents()
  File "/workspaces/ha-core/homeassistant/components/conversation/default_agent.py", line 670, in _rebuild_trigger_intents
    for sentence in intent_data.sentences:
                    ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/functools.py", line 1001, in __get__
    val = self.func(instance)
          ^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/hassil/intents.py", line 52, in sentences
    return [parse_sentence(text, keep_text=True) for text in self.sentence_texts]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/hassil/intents.py", line 52, in <listcomp>
    return [parse_sentence(text, keep_text=True) for text in self.sentence_texts]
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/hassil/parse_expression.py", line 207, in parse_sentence
    seq = parse_expression(chunk, metadata=metadata)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/hassil/parse_expression.py", line 159, in parse_expression
    return parse_group_or_alt_or_perm(chunk, metadata=metadata)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/hassil/parse_expression.py", line 82, in parse_group_or_alt_or_perm
    seq_text = remove_delimiters(seq_chunk.text, GROUP_START, GROUP_END)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/hassil/parser.py", line 302, in remove_delimiters
    assert len(text) > 2, "Text is too short"
           ^^^^^^^^^^^^^
AssertionError: Text is too short
```


### Additional information

_No response_

--- a/homeassistant/components/conversation/trigger.py
+++ b/homeassistant/components/conversation/trigger.py
@@ -26,11 +26,23 @@ def has_no_punctuation(value: list[str]) -> list[str]:
     return value
 
 
+def has_one_non_empty_item(value: list[str]) -> list[str]:
+    """Validate result has at least one item."""
+    if len(value) < 1:
+        raise vol.Invalid("at least one sentence is required")
+
+    for sentence in value:
+        if not sentence:
+            raise vol.Invalid(f"sentence too short: '{sentence}'")
+
+    return value
+
+
 TRIGGER_SCHEMA = cv.TRIGGER_BASE_SCHEMA.extend(
     {
         vol.Required(CONF_PLATFORM): DOMAIN,
         vol.Required(CONF_COMMAND): vol.All(
-            cv.ensure_list, [cv.string], has_no_punctuation
+            cv.ensure_list, [cv.string], has_one_non_empty_item, has_no_punctuation
         ),
     }
 )

--- a/tests/components/conversation/test_trigger.py
+++ b/tests/components/conversation/test_trigger.py
@@ -194,6 +194,27 @@ async def test_fails_on_punctuation(hass: HomeAssistant, command: str) -> None:
         )
 
 
+@pytest.mark.parametrize(
+    "command",
+    [""],
+)
+async def test_fails_on_empty(hass: HomeAssistant, command: str) -> None:
+    """Test that validation fails when sentences are empty."""
+    with pytest.raises(vol.Invalid):
+        await trigger.async_validate_trigger_config(
+            hass,
+            [
+                {
+                    "id": "trigger1",
+                    "platform": "conversation",
+                    "command": [
+                        command,
+                    ],
+                },
+            ],
+        )
+
+
 async def test_wildcards(hass: HomeAssistant, calls, setup_comp) -> None:
     """Test wildcards in trigger sentences."""
     assert await async_setup_component(

--- a/tests/components/conversation/test_trigger.py
+++ b/tests/components/conversation/test_trigger.py
@@ -215,6 +215,22 @@ async def test_fails_on_empty(hass: HomeAssistant, command: str) -> None:
         )
 
 
+async def test_fails_on_no_sentences(hass: HomeAssistant) -> None:
+    """Test that validation fails when no sentences are provided."""
+    with pytest.raises(vol.Invalid):
+        await trigger.async_validate_trigger_config(
+            hass,
+            [
+                {
+                    "id": "trigger1",
+                    "platform": "conversation",
+                    "command": [],
+                },
+            ],
+        )
+
+
+
 async def test_wildcards(hass: HomeAssistant, calls, setup_comp) -> None:
     """Test wildcards in trigger sentences."""
     assert await async_setup_component(

--- a/tests/components/conversation/test_trigger.py
+++ b/tests/components/conversation/test_trigger.py
@@ -230,7 +230,6 @@ async def test_fails_on_no_sentences(hass: HomeAssistant) -> None:
         )
 
 
-
 async def test_wildcards(hass: HomeAssistant, calls, setup_comp) -> None:
     """Test wildcards in trigger sentences."""
     assert await async_setup_component(

